packages/babel-plugin-transform-classes/src/vanilla.js

```diff
@@ -612,4 +622,20 @@ export default class ClassTransformer {
       ),
     );
   }
+
+  /**
+   * For spec mode, add instanceof check for the function block.
+   */
+
+  addNewMethodChecks(map, scope) {
+    const methodRef = scope.generateUidIdentifierBasedOnNode(map._key);
```

Comment:
Don't need to generate one if the method already has an `id`, no?

You do need to check if the name is shadowed inside the method, though, and use scope to rename it.

Reply:
I think there's already logic to generate this name in the non-computed case, but this logic may be running too early for that to have taken affect. We will need logic like this for the computed keys though probably.

Reply:
In that case, I suppose it's best to call the `helper-function-name` first; but I'm not sure if this code (or its caller) has access to the `path`...